android/cli/commands/_buildModule.js

```diff
@@ -1431,30 +1360,52 @@ AndroidModuleBuilder.prototype.packageZip = function (next) {
 				this.logger.info(__('Creating module zip'));
 
 				// 1. documentation folder
-				this.documentation.forEach(function (item) {
-					var fileName = Object.keys(item),
-						content = item[fileName],
-						filePath;
+				var mdRegExp = /\.md$/;
+				(function walk(dir, parent) {
+					if (!fs.existsSync(dir)) return;
+
+					fs.readdirSync(dir).forEach(function (name) {
+						var file = path.join(dir, name);
+						if (!fs.existsSync(file)) return;
+						if (fs.statSync(file).isDirectory()) {
+							return walk(file, path.join(parent, name));
+						}
 
-					fileName = fileName.toString().replace('.md', '.html');
-					filePath = path.join(moduleFolder, 'documentation', fileName);
+						var contents = fs.readFileSync(file).toString();
 
-					dest.append(content, { name: filePath });
+						if (mdRegExp.test(name)) {
+							contents = markdown.toHTML(contents);
+							name = name.replace(/\.md$/, '.html');
+						}
 
-				}, this);
+						dest.append(contents, { name: path.join(parent, name) });
+					});
+				}(this.documentationDir, path.join(moduleFolder, 'documentation')));
 
+				// 2. example folder
 				this.dirWalker(this.exampleDir, function (file) {
 					dest.append(fs.createReadStream(file), { name: path.join(moduleFolder, 'example', path.relative(this.exampleDir, file)) });
 				}.bind(this));
 
+				// 3. platform folder
 				if (fs.existsSync(this.platformDir)) {
 					this.dirWalker(this.platformDir, function (file) {
 						dest.append(fs.createReadStream(file), { name: path.join(moduleFolder, 'platform', path.relative(this.platformDir, file)) });
 					}.bind(this));
 				}
 
+				// 4. Resources folder
+				if (fs.existsSync(this.resourcesDir)) {
+					this.dirWalker(this.resourcesDir, function (file, name) {
+						if (name !== 'README.md') {
+							dest.append(fs.createReadStream(file), { name: path.join(moduleFolder, 'Resources', path.relative(this.resourcesDir, file)) });
+						}
+					}.bind(this));
+				}
+
+				// 5. assets folder, not including js files
 				this.dirWalker(this.assetsDir, function (file) {
-					if (path.extname(file) != '.js' && path.basename(file) != 'README') {
+					if (path.extname(file) !== '.js' && path.basename(file) != 'README') {
```

Comment:
Fixed.

I also fixed an issue in the `validate()` step where an async call was not made async.